mathics/builtin/assignments/internals.py

```diff
@@ -122,6 +124,32 @@ def is_protected(tag, defin):
     return A_PROTECTED & defin.get_attributes(tag)
 
 
+def normalize_lhs(lhs, evaluation):
+    """
+    Process the lhs in a way that
+    * if it is a conditional expression, reduce it to
+      a shallow conditional expression
+      ( Conditional[Conditional[...],tst] -> Conditional[uncondlhs, tst])
+      with `uncondlhs` the result of strip all the conditions from lhs.
+    * if `uncondlhs` is not a `List` or a `Part` expression, evaluate the
```

Comment:
A small note here. I have been using the RsT tagging inside docstrings like this. The reason id this that this is what Sphinx uses by default when told to create system-level documentation.

Reply:
Small thing, as a variable name `lhs` is fine, when not a variable symbol, I think caps is better: Process the LHS because it is in fact an acronym like IBM.

Reply:
the name "uncondlhs" feels awkward. Please consider using underscores to separate words, e.g. "uncond_lhs" or the maybe "stripped_lhs".